parastell/magnet_coils.py

```diff
@@ -401,164 +319,104 @@ class MagnetCoil(object):
-
-        # Extract initial path point
-        pos = cubit.vertex(path_origin).coordinates()
-
-        # Project position vector onto cross-section
-        pos_proj = normalize(pos - np.inner(pos, norm) * norm)
-
-        # Compute angle by which to rotate cross-section such that it faces the
-        # origin
-        rot_ang_orig = np.arccos(np.inner(pos_proj, t_vec_rot))
-
-        # Re-orient rotated cross-section such that thickness vector faces
-        # origin
-        cubit.cmd(
-            f"rotate Surface {surf_id} angle {np.rad2deg(rot_ang_orig)} about "
-            "origin 0 0 0 direction " + " ".join(str(i) for i in norm)
+    @property
+    def coords(self):
+        return self._coords
+
+    @coords.setter
+    def coords(self, data):
+        # Compute tangents
+        tangents = np.subtract(
+            np.append(data[1:], [data[1]], axis=0),
+            np.append([data[-2]], data[0:-1], axis=0),
         )
+        tangents = tangents / np.linalg.norm(tangents, axis=1)[:, np.newaxis]
+
+        # Sample filament coordinates and tangents by modifier
+        self._coords = data[0 : -1 : self.sample_mod]
+        self._coords = np.append(self._coords, [self._coords[0]], axis=0)
+        self.tangents = tangents[0 : -1 : self.sample_mod]
+        self.tangents = np.append(self.tangents, [self.tangents[0]], axis=0)
 
     def create_magnet(self):
-        """Creates magnet coil volumes in cubit.
+        """Creates a single magnet coil CAD solid in CadQuery.
 
         Returns:
-            volume_id (int): magnet volume ids in cubit
+            coil (object): cq.Solid object representing a single magnet coil.
         """
-        # Cross-section inititally populated with thickness vector
-        # oriented along x axis
-        t_vec = np.array([1, 0, 0])
-
-        # Create cross-section for sweep
-        cubit.cmd(f"create surface " + self.shape_str + " zplane")
+        tangent_vectors = [
+            cq.Vector(tuple(tangent)) for tangent in self.tangents
+        ]
 
-        # Store cross-section index
-        cs_id = cubit.get_last_id("surface")
-        # Cross-section initially populated with normal oriented along z
-        # axis
-        cs_axis = np.array([0, 0, 1])
+        # Define coil filament path normals such that they face the filament
+        # center of mass
+        normal_dirs = self._coords - self.center_of_mass
+        normal_dirs = (
+            normal_dirs / np.linalg.norm(normal_dirs, axis=1)[:, np.newaxis]
+        )
 
-        # Initialize path list
-        path = []
+        # Project normal directions onto desired coil cross-section (CS) plane
+        # at each filament position to define true filament path normals
+        parallel_parts = []
+        for dir, tangent in zip(normal_dirs, self.tangents):
+            parallel_parts.append(np.dot(dir, tangent))
+        parallel_parts = np.array(parallel_parts)
```

Comment:
Isn't this the same as
```suggestion
        parallel_parts = np.matmul(normal_dirs, self.tangents.transpose())
```

Reply:
Almost. We'd then need to take the diagonal of the resultant matrix (`parallel_parts` is supposed to be an array of scalars and `np.matmul(normal_dirs, self.tangents.transpose())` results in an NxN matrix where N is the number of points on the filament), but that's easy enough to do.